pkg/log/log.go

```diff
@@ -76,9 +78,15 @@ func (l Logger) WithFields(fields ...zap.Field) Logger {
 // ErrorFilterContextCanceled wraps Logger.Error() and will filter error log when error is context.Canceled
 func (l Logger) ErrorFilterContextCanceled(msg string, fields ...zap.Field) {
 	for _, field := range fields {
-		if field.Key == "error" && field.String == context.Canceled.Error() {
+		if field.Key == "error" && strings.Contains(field.String, context.Canceled.Error()) {
 			return
 		}
+		if field.Type == zapcore.ErrorType {
```

Comment:
1. How about using a `switch` to check `filed.Type` first?
2. Can you add a unit test case for this? ref <https://github.com/pingcap/tidb-lightning/blob/605760d1b2025d1e1a8b7d0c668c74863d7d1271/lightning/log/log_test.go#L43>.